File: WorldFeed/Server/src/Services/Post/WorldFeed.Post.API/Exceptions/ITwitterException.cs
namespace WorldFeed.Post.API.Exceptions
{
    using System;
    using System.Net;

    using WorldFeed.Common.Exceptions;
    using WorldFeed.Upload.Application.Query;

    public interface ITwitterException
    {
        WebException WebException { get; }

        string URL { get; }

        int StatusCode { get; }

        string TwitterDescription { get; }

        DateTime CreationDate { get; }

        ITwitterExceptionInfo[] TwitterExceptionInfos { get; }

        ITwitterQuery TwitterQuery { get; }
    }
}

File: WorldFeed/Client/src/app/core/Core/Exceptions/ITwitterException.ts
import {ITwitterExceptionInfo} from "./ITwitterExceptionInfo";
import {ITwitterQuery} from "../../Public/Models/Interfaces/ITwitterQuery";
import DateTime from "../../../c#-objects/TypeScript.NET-Core/packages/Core/source/Time/DateTime";

export interface ITwitterException {
  webException: WebException;

  URL: string;
  statusCode: number;
  twitterDescription: string;
  creationDate: DateTime;  // DateTimeOffset;
  twitterExceptionInfos: ITwitterExceptionInfo[];
  twitterQuery: ITwitterQuery;
}
